### PR TITLE
Add error pages

### PR DESCRIPTION
**Pull request by brishtibheja**
Added three FAQ pages to the manual. One comes from tenderapp knowledge base that never got ported. I've mostly copy pasted, will do some cleanup later.

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -32,7 +32,10 @@
   - [Field Replacements](templates/fields.md)
   - [Card Generation](templates/generation.md)
   - [Styling & HTML](templates/styling.md)
-  - [Checks and Errors](templates/errors.md)
+  - [Checks and Errors](templates/errors/errors.md)
+    - [Field Replacement Errors](templates/errors/replacement-errors.md)
+    - [Cloze Errors](templates/errors/cloze-error.md)
+    - [Empty Cards](templates/errors/empty-error.md)
```
Comment (dae) [inline src/SUMMARY.md:38]: These filenames are in an errors/ folder, so including error in them is redundant. How about e.g.

templates/errors/field-replacements.md
templates/errors/cloze-deletions.md
templates/errors/empty-cards.md

File: src/templates/errors/empty-error.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Anki displays cards by combining the fields you've entered with a
```
Comment (dae) [inline src/templates/errors/empty-error.md:1]: A heading label would be nice

Comment (brishtibheja) [reply src/templates/errors/empty-error.md]: done!

**Comment by dae** [comment]
Thanks!

**Comment by dae** [comment]
Could I trouble you to adjust these problems in a follow-up commit? I did not notice them until after merging.

```
error: File not found: ../getting-started.md#key-concepts
   ┌─ templates/errors/errors.md:17:12
   │
17 │ Please see [Key Concepts](../getting-started.md#key-concepts) before reading further.
   │            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ../getting-started.md#key-concepts

error: File not found: ./intro.md#the-templates-screen
   ┌─ templates/errors/errors.md:25:72
   │
25 │ - Click the Cards... button at the top of the editing area to open the [templates screen](./intro.md#the-templates-screen).
   │                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./intro.md#the-templates-screen

error: File not found: ./fields.md
   ┌─ templates/errors/errors.md:31:56
   │
31 │ This kind of error indicates an incorrect usage of the [field replacement](./fields.md)
   │                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./fields.md

error: File not found: ./intro.md#the-templates-screen
   ┌─ templates/errors/errors.md:32:[19](https://github.com/ankitects/anki-manual/actions/runs/11174913720/job/31065367403#step:3:20)
   │
32 │ syntax. Visit the [templates screen](./intro.md#the-templates-screen) for the
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./intro.md#the-templates-screen

error: File not found: ./intro.md#the-templates-screen
   ┌─ templates/errors/errors.md:43:23
   │
43 │ To fix this, open the [templates screen](./intro.md#the-templates-screen), and
   │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./intro.md#the-templates-screen

error: File not found: ./fields.md
   ┌─ templates/errors/errors.md:50:[20](https://github.com/ankitects/anki-manual/actions/runs/11174913720/job/31065367403#step:3:21)
   │
50 │ If you don't use a [field replacement](./fields.md) in the front template of a
   │                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./fields.md

error: File not found: ./intro.md#the-templates-screen
   ┌─ templates/errors/errors.md:55:32
   │
55 │ To resolve the issue, open the [templates screen](./intro.md#the-templates-screen),
   │                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ./intro.md#the-templates-screen

error: File not found: ../editing.md#cloze-deletion
   ┌─ templates/errors/errors.md:60:5
   │
60 │ The [cloze](../editing.md#cloze-deletion) filter (as in `{{cloze:Some Field}}`)
   │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ../editing.md#cloze-deletion

error: File not found: ../editing.md#cloze-deletion
   ┌─ templates/errors/errors.md:76:59
   │
76 │ A Cloze notetype's front and back templates should have a [cloze](../editing.md#cloze-deletion)
   │                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ../editing.md#cloze-deletion

warning: Absolute link should be made relative
    ┌─ deck-options.md:167:44
    │
167 │ Anki includes any learning cards that have [crossed the day boundary](/preferences.md#scheduler) (interday learning cards) in the review count,
    │                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Absolute link should be made relative
    │
    = When viewing a document directly from the file system and click on an
      absolute link (e.g. `/index.md`), the browser will try to navigate to
      `/index.md` on the current file system (i.e. the `index.md` file inside
      `/` or `C:\`) instead of the `index.md` file at book's base directory as
      intended.
      
      This warning helps avoid the situation where everything will seem to work
      fine when viewed using a web server (e.g. GitHub Pages or `mdbook serve`),
      but users viewing the book from the file system may encounter broken links.
      
      To ignore this warning, you can edit `book.toml` and set the warning policy to
      "ignore".
      
          [output.linkcheck]
          warning-policy = "ignore"
      
      For more details, see https://github.com/Michael-F-Bryan/mdbook-linkcheck/issues/33
    = Suggestion: change the link to "preferences.md#scheduler"

warning: Absolute link should be made relative
    ┌─ deck-options.md:[22](https://github.com/ankitects/anki-manual/actions/runs/11174913720/job/31065367403#step:3:23)5:40
    │
225 │ Anki treats small steps and steps that [cross a day boundary](/preferences.md#review) differently.
    │                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Absolute link should be made relative
    │
    = Suggestion: change the link to "preferences.md#review"

warning: Absolute link should be made relative
    ┌─ deck-options.md:476:52
    │
476 │ decks using the current preset. You can optionally [adjust the search](/searching.md)
    │                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Absolute link should be made relative
    │
    = Suggestion: change the link to "searching.md"

warning: Absolute link should be made relative
    ┌─ deck-options.md:497:113
    │
497 │ By default, log loss and RMSE (bins) are calculated from all decks using the current preset. You can optionally [adjust the search](/searching.md) before evaluating the parameters, if you'd like to change which cards are used for evaluation.
    │                                                                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Absolute link should be made relative
    │
    = Suggestion: change the link to "searching.md"
```

**Comment by dae** [comment]
Damn, I've just noticed we've broken the existing links inside the clients too. I'm going to revert this change for now, so we can give it some more thought. Can you think of a way to make these changes which doesn't invalidate the existing https://docs.ankiweb.net/templates/errors.html link?

**Pull request merged**